File: origami_master/backend/src/OrigamiMaster.Domain/Entities/ShareLink.cs
using System;

namespace OrigamiMaster.Domain.Entities;

public class ShareLink
{
    public Guid Id { get; set; }
    public Guid CreationId { get; set; }
    public string Token { get; set; } = string.Empty;
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? DisabledAt { get; set; }

    public Creation? Creation { get; set; }

    public void Disable()
    {
        IsActive = false;
        DisabledAt = DateTime.UtcNow;
    }

    public void Toggle()
    {
        IsActive = !IsActive;
        DisabledAt = IsActive ? null : DateTime.UtcNow;
    }
}

File: origami_master/frontend/lib/features/share/domain/entities/share_link.dart
import 'package:equatable/equatable.dart';

class ShareLink extends Equatable {
  final String id;
  final String creationId;
  final String creationName;
  final String shareUrl;
  final String token;
  final DateTime createdAt;
  final bool isActive;

  const ShareLink({
    required this.id,
    required this.creationId,
    required this.creationName,
    required this.shareUrl,
    required this.token,
    required this.createdAt,
    required this.isActive,
  });

  @override
  List<Object?> get props => [
        id,
        creationId,
        creationName,
        shareUrl,
        token,
        createdAt,
        isActive,
      ];
}
